### DISCUSSION TITLE
Clarify AuthenticatorSelection.RequireResidentKey vs ResidentKey

### DISCUSSION THREAD

**Discussion by hochhaus** [go-webauthn/webauthn#89] [Q&A] [answered]
Is anyone able to clarify the difference between `AuthenticatorSelection.RequireResidentKey` and `AuthenticatorSelection.ResidentKey`?

Docs: https://pkg.go.dev/github.com/go-webauthn/webauthn@v0.6.0/protocol#AuthenticatorSelection

Commit: https://github.com/go-webauthn/webauthn/pull/13/commits/d973582f1ae9432913b96511a858ca46618fd1e2

**Comment by james-d-elliott** [comment, marked as answer]
See the spec here: https://www.w3.org/TR/webauthn-2/#dictdef-authenticatorselectioncriteria

Both are valid, however `requireResidentKey` is a webauthn level 1 compatibility layer. Basically you should use `residentKey` unless you KNOW the browser only supports spec level 1.

**Reply by hochhaus** [reply]
Thanks @james-d-elliott. Your link to the spec clarifies it perfectly.

> requireResidentKey ... [Relying Parties](https://www.w3.org/TR/webauthn-2/#relying-party) SHOULD set it to true if, and only if, [residentKey](https://www.w3.org/TR/webauthn-2/#dom-authenticatorselectioncriteria-residentkey) is set to [required](https://www.w3.org/TR/webauthn-2/#dom-residentkeyrequirement-required).
